src/main/java/com/solvd/block2/sql/utilities/DbUtil.java

```diff
@@ -6,24 +6,35 @@
 import java.sql.DriverManager;
 import java.sql.SQLException;
 import java.util.Properties;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
 
 public class DbUtil {
-    private static final String PROPERTIES_FILE = "db.properties";
+    private static final int MAX_CONNECTIONS = 10; // Maximum number of connections in the pool
+    private static final String PROPERTIES_FILE = "db.properties"; // File name for the JDBC properties
+    private static BlockingQueue<Connection> connectionPool;
 
-    private static Connection createConnection() {
+    static {
+        connectionPool = new ArrayBlockingQueue<>(MAX_CONNECTIONS);
+        initializeConnectionPool();
+    }
+
+    private static void initializeConnectionPool() {
         Properties properties = loadProperties();
+
         if (properties != null) {
-            String jdbcUrl = properties.getProperty("jdbc.url");
-            String username = properties.getProperty("jdbc.username");
-            String password = properties.getProperty("jdbc.password");
+            String dbUrl = properties.getProperty("jdbc.url");
+            String dbUser = properties.getProperty("jdbc.username");
+            String dbPassword = properties.getProperty("jdbc.password");
 
             try {
-                return DriverManager.getConnection(jdbcUrl, username, password);
+                for (int i = 0; i < MAX_CONNECTIONS; i++) {
```

Comment:
One thing that I have noticed here is you are initializing your connection pool of size 10 as soon as you call your DbUtil class.

I suggest you improve this by:
- Initializing/updating your connection pool depending on how many connection request you get from clients. (i.e. if 3 connection request are from clients, then your connection pool is of size 3, but it can dynamically scale in size up to 10 depending on how many clients are requesting for a connection.)

Reply:
Fixes applied. Thank you!